<!--
 * Copyright (C) 2013-2025, The Regents of The University of Michigan.
 * All rights reserved.
 * This software was developed in the Biped Lab (https://www.biped.solutions/) 
 * under the direction of Jessy Grizzle, grizzle@umich.edu. This software may 
 * be available under alternative licensing terms; contact the address above.
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 * The views and conclusions contained in the software and documentation are those
 * of the authors and should not be interpreted as representing official policies,
 * either expressed or implied, of the Regents of The University of Michigan.
 * 
 * AUTHOR: Bruce JK Huang (bjhuang[at]umich.edu) and Chenxi Feng (chenxif@umich.edu)
 * WEBSITE: https://www.brucerobot.com/
-->

<launch>
  <arg name="synched_pointcloud_topic" default="/velodyne_points_sync"/>
    <arg name="synched_image_topic" default="/camera/color/image_sync"/>
    <arg name="synched_camerainfo_topic" default="/camera/color/camera_info_sync"/>

  <node pkg="sync_cam_lidar" name="sync_cam_lidar" type="sync_cam_lidar_node" output="screen">
    
    <param name="pointcloud_topic" type="string" value="/velodyne_points" /> 
    <param name="image_topic" type="string" value="/camera/color/image_raw" />
    <param name="camera_info_topic" type="string" value="/camera/color/camera_info" />

    <param name="sync_pointcloud_topic" type="string" value="$(arg synched_pointcloud_topic)" /> 
    <param name="sync_image_topic" type="string" value="$(arg synched_image_topic)" /> 
    <param name="sync_camera_info_topic" type="string" value="$(arg synched_camerainfo_topic)" />

  </node> 
</launch>
